helix-core/src/selection.rs

```diff
@@ -789,11 +783,15 @@ pub fn split_on_matches(
         for mat in regex.find_iter(&fragment) {
             // TODO: retain range direction
             let end = text.byte_to_char(start_byte + mat.start());
-            result.push(Range::new(start, end));
+            let range = Range::new(start, end);
+            // Don't add a zero-width range if a match is at the start of selection
+            if range != Range::point(sel.from()) {
```

Comment:
I wonder if it is enough to test that the range is not a point? Is there any case where we want a selection over a zero-width match?

Reply:
yeah I actually think helix currently handles zero width matches weirdly. For example you can currently use an emply anchor match `^$`  to match empty lines (or `|` to match every char index) in helix. This works for forward search but not reverse search. This behavior is incorrect. I think we should always ignore zero-width matches so that there is always some unambiguous text to select.

There may be an argument that you want to keep search working in that case (since you could view that as selection being a secondary instead of primary effect) but for `s` in particular the primary goal is to select the text which matches the regex pattern and `^$` matches no text (and for consistency and avoiding weird edgecases I think we should do the same for search)